sql/src/main/java/io/crate/analyze/OrderBy.java

```diff
@@ -60,4 +69,83 @@ public boolean isSorted() {
     public void normalize(EvaluatingNormalizer normalizer) {
         normalizer.normalizeInplace(orderBySymbols);
     }
+
+    public boolean hasAggregation() {
+        SortSymbolContext ctx = new SortSymbolContext();
+        for( Symbol symbol : this.orderBySymbols()) {
+            SortSymbolVisitor.INSTANCE.process(symbol, ctx);
+            if(ctx.hasAggregation) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    public static OrderBy fromSymbols(List<Symbol> symbols) {
+        boolean[] reverseFlags = new boolean[symbols.size()];
+        Arrays.fill(reverseFlags, false);
+        Boolean[] nullsFirst = new Boolean[symbols.size()];
+        Arrays.fill(nullsFirst, null);
+        return new OrderBy(symbols, reverseFlags, nullsFirst);
+    }
+
+    public static void toStream(OrderBy orderBy, StreamOutput out) throws IOException {
+        orderBy.writeTo(out);
+    }
+
+    public static OrderBy fromStream(StreamInput in) throws IOException {
+        OrderBy orderBy = new OrderBy();
+        orderBy.readFrom(in);
+        return orderBy;
+    }
+
+    @Override
+    public void readFrom(StreamInput in) throws IOException {
+        int numOrderBy = in.readVInt();
+        reverseFlags = new boolean[numOrderBy];
+
+        for (int i = 0; i < reverseFlags.length; i++) {
+            reverseFlags[i] = in.readBoolean();
+        }
+
+        orderBySymbols = new ArrayList<>(numOrderBy);
+        for (int i = 0; i < reverseFlags.length; i++) {
+            orderBySymbols.add(Symbol.fromStream(in));
+        }
+
+        nullsFirst = new Boolean[numOrderBy];
+        for (int i = 0; i < numOrderBy; i++) {
+            nullsFirst[i] = in.readOptionalBoolean();
+        }
+    }
+
+    @Override
+    public void writeTo(StreamOutput out) throws IOException {
+        out.writeVInt(reverseFlags.length);
+        for (boolean reverseFlag : reverseFlags) {
+            out.writeBoolean(reverseFlag);
+        }
+        for (Symbol symbol : orderBySymbols) {
+            Symbol.toStream(symbol, out);
+        }
+        for (Boolean nullFirst : nullsFirst) {
+            out.writeOptionalBoolean(nullFirst);
+        }
+    }
+
+    private static class SortSymbolContext {
+        public boolean hasAggregation = false;
+    }
+
+    private static class SortSymbolVisitor extends SymbolVisitor<SortSymbolContext, Void> {
```

Comment:
is it okay to not detect aggregations inside a function? 
If so, it could probably be changed to return a Boolean instead of Void, then you wouldn't need a Context anymore